mrmustard/physics/ansatze.py

```diff
@@ -241,6 +259,84 @@ def _order_batch(self):
         self.vec = math.gather(self.vec, sorted_indices, axis=0)
         self.array = math.gather(self.array, sorted_indices, axis=0)
 
+    def decompose_ansatz(self) -> DiffOpPolyExpAnsatz:
+        r"""
+        This method decomposes a DiffOpPolyExpAnsatz. Given an ansatz of dimensions:
+        A=(batch,m+n,m+n), b=(batch,m+n), c = (batch,k_1,k_2,...,k_n),
+        it can be rewritten as an ansatz of dimensions
+        A=(batch,2m,2m), b=(batch,2m), c = (batch,l_1,l_2,...,l_m), with l_i = sum_j k_j
+        This decomposition is typically favourable if n>m, and will only run if that is the case.
```

Comment:
It might be helpful of understanding the code in the future to clarify what are shape_alpha, dim_alpha, shape_beta, dim_beta in this explanation.